roles/kubernetes/master/tasks/kubeadm-certificate.yml

```diff
@@ -0,0 +1,42 @@
+---
+- name: Backup old certs and keys
+  copy:
+    src: "{{ kube_cert_dir }}/{{ item.src }}"
+    dest: "{{ kube_cert_dir }}/{{ item.dest }}"
+    remote_src: yes
+  with_items:
```

Comment:
Does this list need to include everything listed [here](https://kubernetes.io/docs/reference/setup-tools/kubeadm/implementation-details/#constants-and-well-known-values-and-paths)? I don't see `sa.pub`, `sa.key`, `front-proxy-ca.crt`, `front-proxy-ca.key`, `ca.crt` nor `ca.key`.

Reply:
Because of these certificates we can reuse it, it does not affect the connection between apiserver and other components.

Reply:
The most important thing that we did not regenerate ca.key and ca.crt.
Why is there no regeneration? There are two main considerations.
1. We don't need to consider the issue of cluster certificate expiration. The version of k8s1.12 will be updated regularly afterwards. The certificate is also updated when you upgrade the cluster with kubeadm.
2 If you want to regenerate ca.key and ca.crt, you should also consider updating the node node ca.crt when expanding the master. Why not use the previous certificate, it will be easier.

Reply:
The most important thing that we did not regenerate ca.key and ca.crt.
Why is there no regeneration? There are two main considerations.
1. We don't need to consider the issue of cluster certificate expiration. The version of k8s1.12 will be updated regularly afterwards. The certificate is also updated when you upgrade the cluster with kubeadm.
2 If you want to regenerate ca.key and ca.crt, you should also consider updating the node node ca.crt when expanding the master. Why not use the previous certificate, it will be easier.